src/contract.rs

```diff
@@ -592,26 +701,51 @@ pub fn execute_swap(
     // validate input_amount if native input token
     validate_input_amount(&info.funds, input_amount, &input_token.denom)?;
 
-    let token_bought = get_input_price(input_amount, input_token.reserve, output_token.reserve)?;
+    let fees = FEES.load(deps.storage)?;
+    let total_fee_percent = fees.lp_fee_percent + fees.protocol_fee_percent;
+    let token_bought = get_input_price(
+        input_amount,
+        input_token.reserve,
+        output_token.reserve,
+        total_fee_percent,
+    )?;
 
     if min_token > token_bought {
         return Err(ContractError::SwapMinError {
             min: min_token,
             available: token_bought,
         });
     }
+    // Calculate fees
+    let protocol_fee_amount = get_protocol_fee_amount(input_amount, fees.protocol_fee_percent)?;
+    let input_amount_with_protocol_fee = input_amount - protocol_fee_amount;
 
-    // Create transfer from message
-    let mut transfer_msgs = match input_token.denom {
+    // Transfer input amount - protocol fee to contract
```

Comment:
could dump this comment as well if name of variable is more clear. Just a style comment though.